### DISCUSSION TITLE
Sözlük terimlerini oluşturmak için ChatGPT promptu

### DISCUSSION THREAD

**Discussion by emredevsalot** [kamp-us/pano#687] [Brain storming - Fikir Paylaşımı]
Hali hazırda eklenmiş birkaç terimden de yararlanarak diğer terimleri oluşturabilmek için bir prompt yazdım. Altına da onunla oluşturduğum 3 terimi ekliyorum. Kullanabilir miyiz, promptu geliştirebilir miyiz, düşüncelerinizi bekliyorum.

Prompt:
```
Bir sözlük uygulamamız var. Burada yazılım alanında İngilizce terimlerin, Türkçe karşılıklarını açıklıyoruz. Etiket olarak kullanılabilecek terimler de ekliyoruz. Başlık [title] olarak İngilizce, etiketler [tags] olarak İngilizce olacak. Açıklama Türkçe olacak.

[Örnek 1 - Başlangıç]:
---
title: CORS - Cross-Origin Resource Sharing
tags:
  - cors
  - cross-origin-resource-sharing
  - http-access-control
  - web-development
---

Bir web tarayıcısının güvenlik politikalarını kullanarak kaynakların farklı bir sunucudan alınmasına izin veren bir mekanizmadır. Tarayıcıların kötü niyetli web uygulamalarının kaynaklar arasında veri paylaşımını sınırlayarak güvenliği sağlamasına yardımcı olur.
[Örnek 1 - Bitiş]

[Örnek 2 - Başlangıç]:
---
title: Deployment
tags:
  - deployment
  - devops
  - cloud
---

Geliştirilen yazılımın, kullanıcılara sunulması amacıyla yazılımın derlenmesi, yapılandırılması ve hedef ortama yerleştirilmesi adımlarını takip eden, canlı ortama aktarılma işlemidir.
[Örnek 2 - Bitiş]

[Örnek 3 - Başlangıç]:
---
title: UX - User Experience
tags:
  - ux
  - user-experience
  - ui
  - user-interface
---

Kullanıcı deneyimi, bir kullanıcının bir ürün veya hizmetle etkileşimi sırasında yaşadığı deneyimi ifade eder.
[Örnek 3 - Bitiş]


Şimdi benden bir terim isteyeceksin. Ben her bir terim verdiğimde aşağıdaki kurallara uygun olarak title,tags ve açıklama bilgilerini örneklerdeki aynı format ile oluşturacaksın. 

- Yalnızca örneklerin içeriğindeki bilgileri oluştur.
- [Örnek] başlıklarını kullanma.
- Çıktıyı .mdx formatında ver.

Konuşmaya "Lütfen bir terim giriniz." diyerek başla.
```

1. oluşturulan terim - commit:
```
---
title: Commit
tags:
  - commit
  - version-control
  - git
---

Bir versiyon kontrol sistemi (örneğin Git) kullanılırken, dosyaların veya proje kodunun belirli bir anlık durumunu kaydetmek için yapılan işlemdir. Bu, değişikliklerin takip edilmesini, geçmiş versiyonların geri dönüş yapılmasını ve ekip üyeleri arasında işbirliğini kolaylaştırmayı amaçlar.
```

2. oluşturulan terim - accessibility:
```
---
title: Accessibility
tags:
  - accessibility
  - web-development
  - usability
---

Erişilebilirlik, web siteleri, uygulamalar veya diğer dijital ürünlerin, engellilik durumlarına sahip kullanıcılar dahil olmak üzere herkes tarafından kolayca kullanılabilir olma yeteneğini ifade eder. Bu, görme engelli, işitme engelli veya motor beceri kısıtlamaları gibi farklı kullanıcı ihtiyaçlarını karşılamayı amaçlar.
```

3. oluşturulan terim - database:
```
---
title: Database
tags:
  - database
  - data-management
  - storage
---

Veritabanı, yapılandırılmış verilerin depolandığı, yönetildiği ve erişildiği bir elektronik sistemdir. Veritabanları genellikle bilgiyi etkili bir şekilde saklamak, sorgulamak ve güncellemek için kullanılır. Veritabanları çeşitli uygulamalarda, yazılımlarda ve web sitelerinde kullanılır.
```

**Comment by Ketcap** [comment]
Güzel fikir, yalnız direkt ChatGPT'den sözlük yapmak yerine content generation olarak kullanılıp daha sonra üstünden geçilmeli.
Sözlük eğer yanlış değilsem amacı terimleri güzelce açıklamak anlamını translate etmek değil. 

Demek istediğim içeriklerin açıklamasında hala İngilizce terim geçiyor ve onlarda aslında başka yere referans ediyor. Bunu bozmamak.

**Reply by emredevsalot** [reply]
Tabii ki kullanan kişi de üstünden geçip düzenlemeli. Benim amacım bir nevi devtool gibi kullanmak. Mesela ben 10 tane terimi çevirip PR açsam(ChatGPT yardımıyla ya da yardımsız), yine aynı şekilde review'dan geçecek. 

"Tam bunlar gibi istiyoruz" dediğimiz 3-5 veya daha fazla terimi sağladığımızda dil konusunda yeterli olduğunu düşünüyorum. Dilde başka eksik gördüğünüz şeyleri kural olarak ekleyebiliriz.

Hem "başka alanlarda da nasıl kullanabiliriz?" fikirleri oluşur diye paylaşmak istedim, hem de birkaç kişi girişip tüm terim dosyalarını ve ekstra eklemek istediklerimizi çok çabuk bir şekilde doldurabiliriz.

**Reply by Ketcap** [reply]
bu sekilde guzel bir yaklasim development kismini hizlandirmak acisindan. @emredevsalot 1-2 terimi once kendin cevirip daha sonra chatgpt ile cevirip(kendin ustunden gecip) pr acar misin 2 pr'daki icerikleri karsilastiralim farklari nasil.

**Comment by arabacibahadir** [comment]
Sözlük şu an kullanıldığı gibi devam edecekse benim ekleyeceğim ekstra bir şey yok. Belki #563 linter kısmını da gpt'ye hallettirebiliriz. 

Kafamdaki soru işareti ise şu, alanında uzman kişilerin yazı yazdığı bir blog gibi yer olmayacaksa chatgpt'den hep eksik kalacak. Bu noktada daha fazla efor harcanmalı mı emin değilim 🤷‍♂️

**Reply by Ketcap** [reply]
bu sadece icerigi uretmek icin kullanilmasi dusunuluyor sisteme bir kod yazmayacagiz sadece icerikleri yaratirken chatgpt'den yaratip buraya getirme konusu tartisiliyor.

**Reply by emredevsalot** [reply]
evet, şu anki sözlüğün olduğu hali noktasında bunu yapıp, hazır bir hale daha çabuk getirebiliriz. onun dışında "türkçe terimler sözlüğü" olarak açıklaması var README'de.

- bir yandan elimizde böyle bir veritabanı olacağı için, onu pano app'inde de post'larda o kelimeler varsa hover durumunda o açıklamaların gösterilebilmesi gibi bir fikir geldi aklıma. Onu da başka bir discussion'da konuşabiliriz, bu noktada karar kıldıktan sonra.

**Reply by Ketcap** [reply]
@usirin senin kesin boyle bi fikrin yada let's goo diyecegin bi fikir not olmasi icin birakiyorum buraya tag'i

**Reply by usirin** [reply]
Ben bu noktada @arabacibahadir gibi dusunuyorum. Bu icerigin bir bot tarafindan uretilip uretilmemesinden ote karsidaki insanin kolayca anlayabilecegi bir seviyede olmasi gerekiyor. Sozluk'un kurulus vizyonu bu. Tecrubeyi jargonu aradan kaldirmak icin bir kopru olarak kullanmak. Bu vizyon da bu icerigin bir modelden uretilmesini degil de, gerekirse bu icerikten bir model uretilmesini gerektiriyor.

Yani isin ozunde zaten neden acele ediyoruz ki terim uretmek icin? "on demand" generate etmek gibi bir hedefimiz yoksa, bu tarz bir prompt uretip, insanlara gelin bu sekil contribution yapin demek sadece maintenance yukunu arttiracak. Ben acikcasi bu isi boyle yapmamamiz gerektigini dusunuyorum.

Yavas yavas, ozumsenerek, guzel bir review'dan gecerek uretilmis content, "seri uretim" icerikten cok daha onemli bu proje icin.

**Comment by hueypov** [comment]
selamlar, ben şu şekilde denedim. 

```
Prompt : 
Yazılım ve bilgisayar mühendisliği ile alakalı ingilizce terimlerin türkçe karşılıklarını açıkladığımız bir sözlük projemiz var. Ayrıca etiket olarak kullanılabileceğimiz terimler de ekliyoruz. Bana örnek terimler bulmam da yardımcı olabilir misin?

Çıktıyı şu şekilde verebilir misin? 
1) ilk önce açıklanacak olan ingilizce terimi yaz. 
2) Daha sonra altına minimum 4, maksimum 10 adet olacak şekilde ingilizce etiket terimini yaz. 
3) En son olarak başlangıç seviyesi bir dil ile maksimum 150 kelime olacak şekilde türkçe açıklamayı yaz. 

Bana bu şekilde 10 örnek terim verebilir misin?

Output:

1. Variable

Tags: Declaration, Programming, Computer Science, Data Storage
Açıklama: Bir değişken, program içinde veri saklamak için kullanılan sembolik bir isimdir. Değişkenler, sayılar, metinler, boolean (doğru/yanlış) değerler gibi farklı veri türlerini temsil edebilir. Programın farklı noktalarında aynı değeri tutabilirler ve programın akışı sırasında bu değerler değiştirilebilir.

2. Algorithm

Tags: Logic, Problem Solving, Programming, Computation
Açıklama: Bir algoritma, belirli bir problemi çözmek veya belirli bir görevi gerçekleştirmek için adım adım talimatlar içeren bir yönergeler dizisidir. Algoritmalar, hesaplama, veri işleme ve programlama gibi alanlarda yaygın olarak kullanılır. İyi tasarlanmış algoritmalar, verimlilik, doğruluk ve güvenilirlik gibi önemli faktörleri dikkate alır.

3. Object-Oriented Programming (OOP)

Tags: Software Development, Paradigm, Classes, Inheritance
Açıklama: Nesne yönelimli programlama, yazılım geliştirme yaklaşımıdır. Bu yaklaşıma göre, programlar "nesneler" olarak adlandırılan veri yapıları ve onların birbirleriyle etkileşimini içerir. Her nesne, veri ve onunla ilişkili işlevselliği bir araya getiren "sınıflar" tarafından tanımlanır. Kalıtım, sarmalama, çok biçimlilik gibi kavramlar OOP'nin temelini oluşturur.

```
Emre'nin promptunu referans alıp bir kaç kelime ile detay verdim ama emre'nin paylaştığı üzere few-shot prompting ile istediğimiz gibi sonuçlar alınabilir.

**Reply by emredevsalot** [reply]
eline sağlık. sıfırdan kelimeler bulmak için iyi. böyle de yapılabilir.

ama örneklerle öğretmenin daha iyi sonuç vereceğini düşünüyorum. çünkü, 3-4 tane, "tamam bu çok iyi" dediğimiz örneği verirsek, kendimiz dili hakkında bilgi vermek zorunda kalmayız.

çünkü o örneklerden istediğimiz dili, "başlangıç seviyesi" diye belirttiğimiz hâline göre daha iyi predict edebilecek.(https://learnprompting.org/docs/basics/few_shot buna göz atabilirsin denk gelmediysen).

iki türlü de denenir, fark etmez, günün sonunda pr kabul görüp merge'lendiği sürece her türlüsü makul. 3-5 kişi çok kısa sürede çok fazla terim doldurabileceğimiz bir yola girelim istedim.

**Reply by hueypov** [reply]
Evet attığın linkte güzelce açıklanmış. O zaman şuan örnek seçebiliriz. Örneğin şunların kısa ve öz olmalarını beğendim. `---
title: UX - User Experience
tags:
  - ux
  - user-experience
  - ui
  - user-interface
---

Kullanıcı deneyimi, bir kullanıcının bir ürün veya hizmetle etkileşimi sırasında yaşadığı deneyimi ifade eder.`

`---
title: Version Control

tags: 
- Collaboration
- Software Development
- Git
- History
---

Sürüm kontrolü, yazılım geliştirme sürecinde yapılan değişiklikleri izleme, yönetme ve kaydetme pratiğidir. Bu, birden fazla geliştiricinin aynı proje üzerinde çalışmasını kolaylaştırır.`

**Reply by emredevsalot** [reply]
Evet, bu iki örnek de iyiymiş, denediğimde bunları da örnek olarak ekleyeyim. Teşekkür ederim.
